stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/viewholder/ChannelListItemViewHolder.kt

```diff
@@ -0,0 +1,88 @@
+package io.getstream.chat.android.ui.channel.list.adapter.viewholder
+
+import android.annotation.SuppressLint
+import android.util.TypedValue
+import android.view.View
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.ui.channel.list.ChannelListViewStyle
+import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelItemDiff
+import io.getstream.chat.android.ui.databinding.StreamChannelListItemViewBinding
+import io.getstream.chat.android.ui.utils.extensions.context
+import io.getstream.chat.android.ui.utils.extensions.getCurrentUser
+import io.getstream.chat.android.ui.utils.extensions.getCurrentUserRead
+import io.getstream.chat.android.ui.utils.extensions.getDisplayName
+import io.getstream.chat.android.ui.utils.extensions.getLastMessageDisplayText
+import io.getstream.chat.android.ui.utils.extensions.getLastMessageTime
+import io.getstream.chat.android.ui.utils.extensions.getReadStatusDrawable
+import io.getstream.chat.android.ui.utils.extensions.getUsers
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+public class ChannelListItemViewHolder(itemView: View) : BaseChannelListItemViewHolder(itemView) {
+
+    public companion object {
+        @SuppressLint("ConstantLocale")
+        private val DEFAULT_LOCALE: Locale = Locale.getDefault()
+
+        private val TIME_FORMAT = SimpleDateFormat("HH:mm", DEFAULT_LOCALE)
+    }
+
+    public override fun bind(channel: Channel, position: Int, diff: ChannelItemDiff?) {
+        StreamChannelListItemViewBinding.bind(itemView).apply {
+            diff?.run {
+                if (nameChanged) {
+                    channelNameLabel.text = channel.getDisplayName()
+                }
+
+                if (avatarViewChanged) {
+                    avatarView.setChannelData(channel, channel.getUsers())
+                    avatarView.setOnClickListener {
+                        when (channel.getUsers().size) {
+                            1 -> userClickListener?.onUserClick(channel.getCurrentUser())
+                            else -> channelClickListener?.onClick(channel)
+                        }
+                    }
+                }
+
+                if (lastMessageChanged) {
+                    lastMessageLabel.text = channel.getLastMessageDisplayText()
+                    channel.getLastMessageTime()
+                        ?.let(TIME_FORMAT::format)
+                        ?.let { lastMessageTimeLabel.text = it }
+
+                    channel.getCurrentUserRead()?.unreadMessages?.let { unreadCount ->
+//                        BadgeDrawable.create(context).apply {
```

Comment:
What is it? Should it be deleted?

Reply:
This is still a WIP. Only made the PR so we could get the full flow in place.